instrumentation-api/build.gradle

```diff
@@ -1,34 +1,22 @@
 plugins {
-  id 'org.xbib.gradle.plugin.jflex' version '1.5.0'
+  id("org.xbib.gradle.plugin.jflex") version "1.5.0"
+
   id("otel.java-conventions")
   id("otel.jacoco-conventions")
   id("otel.publish-conventions")
 }
 
-group = 'io.opentelemetry.instrumentation'
-
-def jflexTargetDir = file"${project.buildDir}/generated/jflex/sql"
-
-def sqlSanitizerJflex = tasks.register("sqlSanitizerJflex", org.xbib.gradle.plugin.JFlexTask) {
-  group = 'jflex'
-  description = 'Generate SqlSanitizer'
-  source = [file("${project.projectDir}/src/main/jflex/SqlSanitizer.flex")]
-  target = jflexTargetDir
-}
-
-tasks.named("compileJava").configure {
-  dependsOn(sqlSanitizerJflex)
-}
-
-tasks.named("javadoc").configure {
-  dependsOn(sqlSanitizerJflex)
-}
-
-tasks.named("sourcesJar").configure {
-  dependsOn(sqlSanitizerJflex)
+sourceSets {
+  main {
+    java {
+      // gradle-jflex-plugin has a bug in that it always looks for the last srcDir in this source
+      // set to generate into. By default it would be the src/main directory itself.
+      srcDir("${buildDir}/generated/sources/jflex")
```

Comment:
I went ahead and changed the directory a bit to match what gradle-jflex-plugin wanted to use as a default

https://github.com/jprante/gradle-plugin-jflex/blob/master/src/main/groovy/org/xbib/gradle/plugin/JFlexPlugin.groovy#L71

Even though that code can basically never use that default as the java sourceset will always have a srcDir, the src/main dir.